File: 删除排序数组中的重复项 II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        count = 0
        lastPointer = 0
        while lastPointer < len(nums):
            for i in range(lastPointer, len(nums)):
                if nums[lastPointer] == nums[i]:
                    count += 1
            
            if count > 2:
                while (count - 2) != 0:
                    del nums[lastPointer]
                    count -= 1
                    
                count = 0
                lastPointer += 2
            
            else:
                count = 0
                lastPointer += 1
    
        return lastPointer

File: 删除排序数组中的重复项 II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int len = nums.size();
        if(len <= 2)
        {
            return len;
        }
        int pointer = 1;
        for(int i = 2; i < n; i++)
        {
            if(nums[i] != nums[pointer - 1])
            {
                nums[++pointer] = nums[i];
            }
        }
        return pointer + 1;
    }
};
